classes/external/search_packages.php

```diff
@@ -284,11 +293,36 @@ private static function create_recently_used_sql(int $contextid): array {
         return [$joinlastusedsql, ['contextid' => $contextid]];
     }
 
+    /**
+     * Construct a conjunctive where clause.
+     *
+     * @param string ...$clauses
+     * @return string
+     */
+    private static function sql_where(string ...$clauses): string {
+        $where = '';
+
+        foreach ($clauses as $clause) {
+            if ($clause === '') {
+                continue;
+            }
+            $where .= "({$clause}) AND ";
+        }
+        if ($where !== '') {
+            // Remove last ' AND '.
+            $where = substr($where, 0, -5);
+            $where = "WHERE $where";
+        }
```

Comment:
Das geht kompakter mit array_filter und implode (nicht getestet):
```suggestion
        $where = implode(') AND (', array_filter($clauses, 'strlen'));
        if ($where !== '') {
            $where = "WHERE ($where)";
        }
```